### PR TITLE
ssl-proxy: init at 0.2.7

### PR DESCRIPTION
**Pull request by konst-aa**
## Description of changes

Packaged `ssl-proxy`.
Homepage: https://github.com/suyashkumar/ssl-proxy
Description taken from the repo: [ssl-proxy is a] simple single-command SSL reverse proxy with autogenerated certificates (LetsEncrypt, self-signed).

## Things done

Just made a `default.nix` under `pkgs/tools/networking/ssl-proxy` that uses `buildGo118Module`

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    homepage = "https://github.com/suyashkumar/ssl-proxy";
+    description = "Simple single-command SSL reverse proxy with autogenerated certificates (LetsEncrypt, self-signed)";
+    longDescription = ''
+          A handy and simple way to add SSL to your thing running on a VM--be it your personal jupyter
+          notebook or your team jenkins instance. ssl-proxy autogenerates SSL certs and proxies
+          HTTPS traffic to an existing HTTP server in a single command.
+    '';
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [konst-aa];
+    platforms = lib.platforms.linux;
```
Comment (NotAShelf) [inline pkgs/tools/networking/ssl-proxy/default.nix:27]: you can instead put lib in scope under meta

```nix
meta = with lib; {
	license = licenses.mit;
}
```

etc.

**Review by NotAShelf — CHANGES_REQUESTED** [review]
needs minor changes to fit nixpkgs format

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/suyashkumar/ssl-proxy";
+    description = "Simple single-command SSL reverse proxy with autogenerated certificates (LetsEncrypt, self-signed)";
+    longDescription = ''
+      A handy and simple way to add SSL to your thing running on a VM--be it your personal jupyter
+      notebook or your team jenkins instance. ssl-proxy autogenerates SSL certs and proxies
+      HTTPS traffic to an existing HTTP server in a single command.
+    '';
+    license = licenses.mit;
+    maintainers = [ maintaners.konst-aa ];
```
Comment (emilylange) [inline pkgs/tools/networking/ssl-proxy/default.nix:28]:
```suggestion
    maintainers = [ maintainers.konst-aa ];
```

Typo.
Should fix eval.

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/ssl-proxy/default.nix (hunk 2 above)
Comment (konst-aa) [reply pkgs/tools/networking/ssl-proxy/default.nix:28]: https://www.youtube.com/watch?v=iamld0fZkLM

File: pkgs/tools/networking/ssl-proxy/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, buildGo118Module, fetchFromGitHub }:
```
Comment (NotAShelf) [inline pkgs/tools/networking/ssl-proxy/default.nix:1]: are you using stdenv anywhere? if not, can be removed

Comment (konst-aa) [reply pkgs/tools/networking/ssl-proxy/default.nix]: done!

**Comment by NotAShelf** [comment]
could you resolve the merge conflict? probably something popped up when you were rebasing

**Comment by konst-aa** [comment]
oh no

**Pull request closed without merging**